doc/tools/coqrst/coqdomain.py

```diff
@@ -1167,16 +1168,22 @@ def GlossaryDefRole(typ, rawtext, text, lineno, inliner, options={}, content=[])
     #pylint: disable=dangerous-default-value, unused-argument
     env = inliner.document.settings.env
     std = env.domaindata['std']['objects']
-    key = ('term', text)
+    m = re.match(r"([^<]+)<(.+)>", text)
```

Comment:
The regexp you want is `explicit_title_re` in `sphinx.util.docutils.ReferenceRole` ; in fact inheriting from that role would mean you don't even have to do the parsing yourself, I think.

Reply:
I tried to inherit from that role but I don't know why I get this error:

```
  File "/home/jfehrle/.local/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 880, in interpreted
    nodes, messages2 = role_fn(role, rawsource, text, lineno, self)
TypeError: object() takes no parameters
```

With this code:

```
class GlossaryDefRole(ReferenceRole):

    role_name = "gdef"
    
    def __call__(self, typ, rawtext, text, lineno, inliner, options={}, content=[]):
        #pylint: disable=dangerous-default-value, unused-argument
        super().__call__(typ, rawtext, text, lineno, inliner, options={}, content=[])
        env = inliner.document.settings.env
          :
```